source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -89,14 +90,17 @@ namespace aspect
 
         // Noticed that the size of matrix A is n_particles x matrix_dimension
         // which usually is not a square matrix. Therefore, we find the
-        // least squares solution of Ax=r by solving the "normal" equations
-        // (A^TA) x = A^Tr.
+        // least squares solution of Ax=r by solving reduced QR factorization
+        // Ax = QRc = b -> Q^TQRc = Rc =Q^Tb
```

Comment:
In the line above you reference the problem as `Ax=r` while here you use `Ax = b`. Is `b` different from `r` or do you just use a different name to avoid confusion with the `r` matrix? If so, please change the line above into `b` just to be consistent.